### PR TITLE
teswiz assignment 01 : amazon

### PR DESCRIPTION
**Pull request by rsshekhawat**

### CODE REVIEW

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomepageBL.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    public AmazonHomepageBL searchForProduct(String product) {
+        AmazonSearchResultsScreen amazonSearchResultsScreen = AmazonHomeScreen.get().searchProductUsingAmazonSearchBar(product);
+
+        String actualSearchWasFor = amazonSearchResultsScreen.getActualSearchString();
+        softly.assertThat(actualSearchWasFor).as("Search was for a different value").isEqualTo(product);
+
+        int numberOfProductsFound = amazonSearchResultsScreen.getNumberOfProductsFound();
+        softly.assertThat(numberOfProductsFound).as("Insufficient search results retrieved").isGreaterThan(0);
+        return this;
```
Comment (mohammadsrk) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomepageBL.java:44]: Why this is a soft assert?

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ShoppingCartBL.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        boolean isProductPresent = false;
+
+        for(String productTitle: productTitles){
+            if(productTitle.toLowerCase().contains(product)) {
+                isProductPresent = true;
+                break;
+            }
+        }
+        Assert.assertTrue(isProductPresent, "The product is not present in shopping cart");
+        return this;
```
Comment (mohammadsrk) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ShoppingCartBL.java:49]: Missing loggers

File: src/test/java/com/znsio/sample/e2e/screen/android/amazon/AmazonHomeScreenAndroid.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+public class AmazonHomeScreenAndroid extends AmazonHomeScreen {
+    private final Driver driver;
+    private final Visual visually;
+    private static final String SCREEN_NAME = AmazonHomeScreenAndroid.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+    private static final String NOT_YET_IMPLEMENTED = " not yet implemented";
+    private static final By searchBoxByClass = By.className("android.widget.EditText");
+    private static final By searchIconByXpath = By.xpath("//android.view.View[@resource-id=\"nav-search-form\"]//android.widget.Button");
+
```
Comment (mohammadsrk) [inline src/test/java/com/znsio/sample/e2e/screen/android/amazon/AmazonHomeScreenAndroid.java:19]: The naming convention for xpath is incorrect.

File: caps/ajio_local_capabilities.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "android": {
     "adbExecTimeout": 45000,
     "app": {
-      "local": "https://github.com/anandbagmar/sampleAppsForNativeMobileAutomation/raw/main/ajio-8-3-4.apk"
+      "local": "/Users/rahul2.shekhawat/Desktop/Misc/APKs/ajio-8-3-4.apk"
```
Comment (vishal07111998) [inline caps/ajio_local_capabilities.json:5]: dynamic path should be there

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            if (productTitle.toLowerCase().contains(product)) {
+                isProductPresentInSearchResults = true;
+                break;
+            }
+        }
+
+        Assert.assertTrue(isProductPresentInSearchResults, "Search results do not contain searched product");
+
+        /**
+         String actualSearchWasFor = amazonSearchResultsScreen.getActualSearchString();
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java:58]: remove commented code

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+         int numberOfProductsFound = amazonSearchResultsScreen.getNumberOfProductsFound();
+         softly.assertThat(numberOfProductsFound).as("Insufficient search results retrieved").isGreaterThan(0);
+         */
+        return this;
+    }
+
+    public AmazonHomePageBL selectFirstProduct() {
+
+        LOGGER.info("Selecting the product from the search results");
+        AmazonSearchResultsScreen amazonSearchResultsScreen = AmazonSearchResultsScreen.get();
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java:69]: chaining can be used here

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazon.AmazonProductViewPageScreen;
+import com.znsio.sample.e2e.screen.amazon.AmazonShoppingCartScreen;
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.Assert;
+
+public class ProductViewBL {
+    private static final Logger LOGGER = Logger.getLogger(AmazonHomePageBL.class.getName());
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java:14]: BL name is not same as Logger name

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java:16]: remove unused variable

File: build.gradle [hunk 9]
```diff
@@ -54,7 +54,7 @@ task run(type: JavaExec) {
         def logDirectory = "target/" + getCurrentDatestamp() + "/" + getCurrentTimestamp()
         println "Using LOG_DIR: $logDirectory"
         System.setProperty "LOG_DIR", "$logDirectory"
-        environment "APPLITOOLS_API_KEY", System.getenv("teswiz_APPLITOOLS_API_KEY")
+        environment "APPLITOOLS_API_KEY", "gWD186hRDxALn6wle7MjkPWAX3RyrLwx5rCri5o73Gk110"
```
Comment (vishal07111998) [inline build.gradle:57]: applitools api key can be dynamic on the basis of the system on which it is running

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.Assert;
+
+import java.util.List;
+
+public class AmazonHomePageBL {
+    private static final Logger LOGGER = Logger.getLogger(AmazonHomePageBL.class.getName());
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/AmazonHomePageBL.java:18]: remove unused variable

File: src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.GUEST_USER;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public ProductViewBL createShoppingCart() {
+
+        AmazonProductViewPageScreen amazonProductViewPageScreen = AmazonProductViewPageScreen.get().addProductToCart();
+        String expectedSuccessMessage = SAMPLE_TEST_CONTEXT.ADD_TO_CART_SUCCESS_MESSAGE;
+        String actualSuccessMessage = amazonProductViewPageScreen.getAddToCartSuccessMessage().toLowerCase();
+        LOGGER.info(System.out.printf("Add To Cart Success Message : %s", actualSuccessMessage));
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/businessLayer/amazon/ProductViewBL.java:42]: no need to add sysout in LOGGERs

File: src/test/java/com/znsio/sample/e2e/screen/web/amazon/AmazonSearchResultsScreenWeb.java [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+        return productInSearchString;
+    }
+
+    @Override
+    public AmazonProductViewPageScreenWeb clickOnFirstProductInSearchResultsList() {
+
+        LOGGER.info("Clicking on first product in search results list");
+        WebElement firstProduct = driver.waitTillElementIsPresent(firstProductByXpath);
+        firstProduct.click();
+        // driver.switchToNextTab();
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/web/amazon/AmazonSearchResultsScreenWeb.java:60]: remove commented code

File: src/test/java/com/znsio/sample/e2e/screen/web/amazon/AmazonShoppingCartScreenWeb.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        LOGGER.info(String.format("Shopping Cart Page Title : %s", shoppingCartPageTitle));
+        return shoppingCartPageTitle;
+    }
+
+    @Override
+    public List<String> getTitleOfAllProductsInShoppingCart() {
+
+        LOGGER.info("Fetching titles of all products in the shopping cart");
+        WebDriverWait wait = new WebDriverWait(driver.getInnerDriver(),30);
+        List<WebElement> productTitleWebElements = wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(shoppingCartProductTitlesByCSS));
```
Comment (vishal07111998) [inline src/test/java/com/znsio/sample/e2e/screen/web/amazon/AmazonShoppingCartScreenWeb.java:45]: there is prebuilt dynamic waits, prefer those

**Pull request closed without merging**